CMakeLists.txt

```diff
@@ -193,50 +194,53 @@ endif()
 
 configure_file(scripts/pugixml.pc.in pugixml.pc @ONLY)
 
-if (NOT DEFINED PUGIXML_RUNTIME_COMPONENT)
-  set(PUGIXML_RUNTIME_COMPONENT Runtime)
-endif()
-
-if (NOT DEFINED PUGIXML_LIBRARY_COMPONENT)
-  set(PUGIXML_LIBRARY_COMPONENT Library)
-endif()
-
-if (NOT DEFINED PUGIXML_DEVELOPMENT_COMPONENT)
-  set(PUGIXML_DEVELOPMENT_COMPONENT Development)
-endif()
-
-set(namelink-component)
-if (NOT CMAKE_VERSION VERSION_LESS 3.12)
-  set(namelink-component NAMELINK_COMPONENT ${PUGIXML_DEVELOPMENT_COMPONENT})
-endif()
-install(TARGETS ${install-targets}
-  EXPORT pugixml-targets
-  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT ${PUGIXML_RUNTIME_COMPONENT}
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT ${PUGIXML_LIBRARY_COMPONENT} ${namelink-component}
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT ${PUGIXML_DEVELOPMENT_COMPONENT}
-  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}${versioned-dir})
-
-install(EXPORT pugixml-targets
+export(TARGETS ${install-targets}
   NAMESPACE pugixml::
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pugixml COMPONENT ${PUGIXML_DEVELOPMENT_COMPONENT})
+  FILE pugixml-targets.cmake)
```

Comment:
@zeux before my change `export` command was used with _export set_:
```cmake
export(EXPORT pugixml-targets
  NAMESPACE pugixml::)
```
From cmake `install` command:
```cmake
install(TARGETS ${install-targets}
  EXPORT pugixml-targets
  ...
```
But now, since `install` rules are under `PUGIXML_INSTALL` condition, we have to change the used signature of `export` command - call with [list of targets](https://cmake.org/cmake/help/latest/command/export.html#exporting-targets) instead of [export set](https://cmake.org/cmake/help/latest/command/export.html#exporting-targets-matching-install-export)